File: LC 365.cpp
class Solution {
public:
    int gcd(int x, int y) {
        while (y) {
            int tmp = y;
            y = x % y;
            x = tmp;
        }
        return x;
    }
    bool canMeasureWater(int jug1Capacity, int jug2Capacity, int targetCapacity) {
        int x = gcd(jug1Capacity, jug2Capacity);
        if (targetCapacity % x == 0 && jug1Capacity + jug2Capacity >= targetCapacity) {
            return true;
        }
        return false;
    }
};

File: 1000/LC 365.go
func canMeasureWater(x int, y int, target int) bool {
    if x + y < target {
        return false
    }

    return target % gcd(x, y) == 0
}

func gcd(x, y int) int {
    for y > 0 {
        x, y = y, x % y
    }

    return x
}